### PR TITLE
Resolve PR comments

### PR DESCRIPTION
**Pull request by groeges**
Signed-off-by: Steven Groeger <groeges@uk.ibm.com>

**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind cleanup

**What does does this PR do / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 1]
```diff
@@ -195,6 +199,7 @@ func (a Adapter) waitForManifestDeployCompletion(applicationName string, gvr sch
 		for {
 			val, ok := <-w.ResultChan()
 			if !ok {
+				klog.V(4).Info("TEST: Watch channel was closed")
```
Comment (EnriqueL8) [inline pkg/devfile/adapters/kubernetes/component/adapter.go]: need to remove this log

File: pkg/devfile/adapters/kubernetes/component/adapter.go [hunk 2]
```diff
@@ -297,6 +298,9 @@ func (a Adapter) Deploy(parameters common.DeployParameters) (err error) {
 				instanceFound = true
 				deploymentManifest.SetResourceVersion(item.GetResourceVersion())
 				deploymentManifest.SetAnnotations(item.GetAnnotations())
+				// If deployment is a `Service` of type `ClusterIP` then the service in the manifest will probabaly not
```
Comment (EnriqueL8) [inline pkg/devfile/adapters/kubernetes/component/adapter.go]: typo: `probabaly`

File: pkg/odo/cli/component/deploy.go [hunk 3]
```diff
@@ -135,9 +134,12 @@ func (do *DeployOptions) Run() (err error) {
 	}
 
 	manifestURL := metadata.Manifest
+	if manifestURL == "" {
+		return errors.New("Unable to deploy as alpha.deployment-manifest is not defined in devfile.yaml")
+	}
```
Comment (EnriqueL8) [inline pkg/odo/cli/component/deploy.go:139]: For another PR, need to verify that this URL is valid

File: pkg/util/util.go [hunk 4]
```diff
@@ -996,8 +996,13 @@ func DownloadFileInMemory(url string) ([]byte, error) {
...
 		return nil, err
 	}
+
 	defer resp.Body.Close()
 
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		return nil, errors.New(http.StatusText(resp.StatusCode))
+	}
+
 	return ioutil.ReadAll(resp.Body)
```
Comment (EnriqueL8) [inline pkg/util/util.go:1006]: Future PR: verify this is not empty

**Review by EnriqueL8 — COMMENTED** [review]
Looks good - very minor comments and some future comments

File: pkg/odo/cli/component/deploy.go (hunk 3 above)
Comment (upLukeWeston) [reply pkg/odo/cli/component/deploy.go:139]: There is a `util.ValidateURL` function already that we can use

Comment (EnriqueL8) [reply pkg/odo/cli/component/deploy.go:139]: amazing

File: pkg/devfile/adapters/kubernetes/component/adapter.go (hunk 1 above)
Comment (groeges) [reply pkg/devfile/adapters/kubernetes/component/adapter.go]: Whoops. Done

File: pkg/devfile/adapters/kubernetes/component/adapter.go (hunk 2 above)
Comment (groeges) [reply pkg/devfile/adapters/kubernetes/component/adapter.go]: Fixed

File: pkg/odo/cli/component/deploy.go (hunk 3 above)
Comment (groeges) [reply pkg/odo/cli/component/deploy.go:139]: Hmmm, doesn't seem to fail when an invalid URL is passed!!!

Comment (groeges) [reply pkg/odo/cli/component/deploy.go:139]: OK, only does basic validation of URL - maybe we can enhance that another time.
Anyway, I have added the check to validate the URL.

**Comment by groeges** [comment]
@EnriqueL8 @upLukeWeston Have updated the PR. Please re-review.

**Review by upLukeWeston — APPROVED** [review]
Looks good to me!

**Review by EnriqueL8 — APPROVED** [review]

**Pull request merged**